pkg/monitoring/perfscale/vmi-phase-transitions.go

```diff
@@ -147,7 +153,31 @@ func updateVMIPhaseTransitionTimeFromCreationTimeHistogramVec(histogramVec *prom
 	}
 
 	histogram.Observe(diffSeconds)
+}
+
+func updateVMIPhaseTransitionTimeFromDeletionTimeHistogramVec(histogramVec *prometheus.HistogramVec, oldVMI *v1.VirtualMachineInstance, newVMI *v1.VirtualMachineInstance) {
+	if !newVMI.IsMarkedForDeletion() || !newVMI.IsFinal() {
+		return
+	}
+
+	if oldVMI == nil || oldVMI.Status.Phase == newVMI.Status.Phase {
+		return
+	}
 
+	diffSeconds, err := getTransitionTimeSeconds(false, true, oldVMI, newVMI)
+	if err != nil {
+		log.Log.V(4).Infof("Error encountered during vmi transition time histogram calculation: %v", err)
+		return
+	}
```

Comment:
there are cases where this won't always be accurate.

For example, a VMI guest can transition to `vmi.Status.Phase = Succeeded` while `vmi.DeletionTimestamp == nil`. There's nothing that prevents the guest from shutting itself down internally.

so the question is, how do you want to handle that scenario? Right now the `getTransitionTimeSeconds` function is going to return `0.0`  in those cases

Reply:
Creation timestamp will catch that, but that doesn't match the intent of the deletion metric.... The creation metric has 1 corner case which is when the `vmi.DeletionTimestamp != nil`, so if it can catch that, then I can accurately report the transition time for the slow deletes.  Let me see if that works.

Reply:
>For example, a VMI guest can transition to vmi.Status.Phase = Succeeded while vmi.DeletionTimestamp == nil. There's nothing that prevents the guest from shutting itself down internally.
>
>so the question is, how do you want to handle that scenario? Right now the getTransitionTimeSeconds function is going to return 0.0 in those cases
>

`0.0` is fine.  The deletion metric will only capture the case where KubeVirt has a deletionTimeStamp on the VMI and the informer sees it.   The creation metric will capture everything else.